Backport of r696900 from the xalan SVN repository
to the xalan-c-1.10.0 source tree.

References:
http://svn.apache.org/viewvc?view=rev&revision=696900
https://bugs.gentoo.org/show_bug.cgi?id=242218

2008-10-18 Martin von Gagern

diff -pur --exclude='*~' c.orig/src/xalanc/Utils/MsgCreator/ICUResHandler.cpp c/src/xalanc/Utils/MsgCreator/ICUResHandler.cpp
--- c.orig/src/xalanc/Utils/MsgCreator/ICUResHandler.cpp	2008-10-18 19:09:22.000000000 +0200
+++ c/src/xalanc/Utils/MsgCreator/ICUResHandler.cpp	2008-10-18 19:13:21.000000000 +0200
@@ -100,15 +100,15 @@ void ICUResHandler::startDocument()
 
 }
 
-void ICUResHandler::endElement(const XMLCh* const ,
-					const XMLCh* const localname,
-					const XMLCh* const )
+void ICUResHandler::endElement(const XMLCh* const /* uri */,
+					const XMLCh* const /* localname */,
+					const XMLCh* const qname)
 {
 	if ( m_startCollectingCharacters == false)
 		return;
 
 
-    if(!XMLString::compareString(localname,s_targetXMLCh))
+    if(!XMLString::compareString(qname,s_targetXMLCh))
 	{
 		m_startCollectingCharacters = false;
 
@@ -122,13 +122,13 @@ void ICUResHandler::startElement(const  
 								const   Attributes&		attributes)
 {
 	
-	if(!XMLString::compareString(localname,s_transUnitXMLCh))
+	if(!XMLString::compareString(qname,s_transUnitXMLCh))
 	{
 		// this is an elemente, SAX2Handler class is responsible to handle:
 		// creating Index file, commom for all localization styles
 		SAX2Handler::startElement(uri, localname, qname, attributes);
 	}
-	else if(!XMLString::compareString(localname,s_targetXMLCh))
+	else if(!XMLString::compareString(qname,s_targetXMLCh))
 	{
 		if ( m_locale != 0 )
 		{
diff -pur --exclude='*~' c.orig/src/xalanc/Utils/MsgCreator/InMemHandler.cpp c/src/xalanc/Utils/MsgCreator/InMemHandler.cpp
--- c.orig/src/xalanc/Utils/MsgCreator/InMemHandler.cpp	2008-10-18 19:09:22.000000000 +0200
+++ c/src/xalanc/Utils/MsgCreator/InMemHandler.cpp	2008-10-18 19:11:11.000000000 +0200
@@ -99,15 +99,15 @@ void InMemHandler::startDocument()
 
 }
 
-void InMemHandler::endElement(const XMLCh* const ,
-					const XMLCh* const localname,
-					const XMLCh* const )
+void InMemHandler::endElement(const XMLCh* const /* uri */,
+					const XMLCh* const /* localname */,
+					const XMLCh* const qname)
 {
 	if ( m_startCollectingCharacters == false)
 		return;
 
 
-	if(!XMLString::compareString(localname,s_targetXMLCh))
+	if(!XMLString::compareString(qname,s_targetXMLCh))
 	{
 		m_startCollectingCharacters = false;
 
@@ -121,13 +121,13 @@ void InMemHandler::startElement(const   
 								const   Attributes&		attributes)
 {
 	
-	if(!XMLString::compareString(localname,s_transUnitXMLCh))
+	if(!XMLString::compareString(qname,s_transUnitXMLCh))
 	{
 		// this is an elemente, SAX2Handler class is responsible to handle:
 		// creating Index file, commom for all localization styles
 		SAX2Handler::startElement(uri, localname, qname, attributes);
 	}
-	else if(!XMLString::compareString(localname,s_targetXMLCh))
+	else if(!XMLString::compareString(qname,s_targetXMLCh))
 	{
 
 		
diff -pur --exclude='*~' c.orig/src/xalanc/Utils/MsgCreator/SAX2Handler.cpp c/src/xalanc/Utils/MsgCreator/SAX2Handler.cpp
--- c.orig/src/xalanc/Utils/MsgCreator/SAX2Handler.cpp	2008-10-18 19:09:22.000000000 +0200
+++ c/src/xalanc/Utils/MsgCreator/SAX2Handler.cpp	2008-10-18 19:12:19.000000000 +0200
@@ -92,12 +92,12 @@ void SAX2Handler::printToIndexFile( cons
 
 
 
-void SAX2Handler::startElement(const   XMLCh* const    ,
-									const   XMLCh* const    localname,
-									const   XMLCh* const    ,
+void SAX2Handler::startElement(const   XMLCh* const    /* uri */,
+									const   XMLCh* const    /* localname */,
+									const   XMLCh* const    qname,
                                     const   Attributes&		attributes)
 {
-	if(!XMLString::compareString(localname,s_transUnitXMLCh))
+	if(!XMLString::compareString(qname,s_transUnitXMLCh))
 	{
 		unsigned int len = attributes.getLength();
 		
